### ISSUE TITLE
VMAP and adserror

### ISSUE THREAD

**Issue by Vladislavik** [googleads/videojs-ima#858]
HI, there is no adserror callback, when in VMAP preroll or midroll doesnot  have any ad, is there is a way to do this callback?

**Comment by arnaudcasame** [comment]
Hi @Vladislavik,

When the ad you're dealing with is a VMAP, to capture the any error event that occurs while requesting, loading or displaying the ad, you can register to the LOG event ( `google.ima.AdEvent.Type.LOG`) and you'll be notified in case an error is thrown during the ad process.

**Comment by Vladislavik** [comment]
> Hi @Vladislavik,
> 
> When the ad you're dealing with is a VMAP, to capture the any error event that occurs while requesting, loading or displaying the ad, you can register to the LOG event ( `google.ima.AdEvent.Type.LOG`) and you'll be notified in case an error is thrown during the ad process.

I am trying to do this:

```
var player=videojs('div');
        var options_ads = {
            adTagUrl: 'URL'
        };
player.ima(options_ads);
player.on("adsready", function(){
                            olplayer.ima.addEventListener(google.ima.AdEvent.Type.LOG, function(event){console.log('ADS LOG: '+event);});
});
```

But there is no ADS LOG event in console, what i am doing wrong?

**Comment by Vladislavik** [comment]
there is error in console? how to fix it?

```
video.min.js:18 VIDEOJS: ERROR: TypeError: Cannot read property '__closure_events_fn_672478766' of undefined
    at gh (ima3.js:98)
    at fh (ima3.js:94)
    at Y.m.addEventListener (ima3.js:98)
    at o.addEventListener (videojs.ima.min.js:1)
    at r.addEventListener (videojs.ima.min.js:1)
    at a.<anonymous> (videojs.ima.min.js:1)
    at b.<anonymous> (embed_player.php?vid=7T7dDZYy59zP&autoplay=no&embed_from=embed_from&http_referer=:648)
    at HTMLDivElement.e (video.min.js:18)
    at HTMLDivElement.d.dispatcher.d.dispatcher (video.min.js:18)
    at Object.k [as trigger] (video.min.js:18)
```

**Comment by Kiro705** [comment]
Hi @Vladislavik ,

The IMA SDK for html5 just released [v3.355.3](https://developers.google.com/interactive-media-ads/docs/sdks/html5/v3/history) which fixes an issue with the LOG event.

Let me know if you are still seeing an issue with the LOG events in your project.

Thanks,
Jackson
IMA SDK DevRel

**Comment by arnaudcasame** [comment]
Hi,

I’m closing this issue because it has been inactive for more than a month. Please reopen if you still encounter this issue with the [latest stable version](https://github.com/googleads/videojs-ima/releases/tag/v1.7.4)

Thank you!
